web/src/components/planner/PlannerFactoryProducts.vue

```diff
@@ -64,6 +64,13 @@
           variant="outlined"
           @input="updateFactory(factory)"
         />
+        <v-btn
+          v-if="!getPart(factory, product.id).satisfied"
```

Comment:
```suggestion
          v-if="!factory.part[product.id]?.satisfied"
```
You can use a null chain operator here to say if the object doesn't exist don't bother trying to get the attribute just return undefined. 

Double check this though.

Reply:
Yeah, I can simplify it that way

Reply:
Updated in latest commit